### PR TITLE
[stable/datadog] Allow providing volumes and mounts in the cluster agent deployment

### PR DESCRIPTION
**Pull request by xornivore**
Signed-off-by: Ivan Ilichev <ivan.ilichev@datadoghq.com>

#### What this PR does / why we need it:

Allow providing volumes and mounts in the cluster agent deployment.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

File: stable/datadog/values.yaml [hunk 1]
```diff
@@ -467,6 +467,22 @@ clusterAgent:
   #
   # useHostNetwork: true
 
+  ## @param volumes - list of objects - optional
```
Comment (clamoriniere) [inline stable/datadog/values.yaml:470]: could you update the Readme.md the new parameters too?

**Review by clamoriniere — CHANGES_REQUESTED** [review]
Hi @xornivore 
Your modification looks fine for me.
Could you add a test case that mounts a volume. To add a test case you just need to add a `values.yaml` file with the parameters you added in the `./ci` folder

File: stable/datadog/ci/cluster-agent-volumes-and-mounts.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  enabled: true
+
+  volumes:
+    - name: secrets
+      secret:
+        secretName: test-secrets
+
+  volumeMounts:
+    - name: secrets
+      mountPath: /etc/secrets
```
Comment (CharlyF) [inline stable/datadog/ci/cluster-agent-volumes-and-mounts.yaml]: Could we also document the `"/etc/datadog-agent/certificates"` path to allow customers to mount certificates for the External Metrics Provider ?
https://github.com/DataDog/datadog-agent/blob/master/cmd/cluster-agent/custommetrics/server.go#L97

Comment (xornivore) [reply stable/datadog/ci/cluster-agent-volumes-and-mounts.yaml]: So this is just a test here. Do you mean should add some more info about possible use cases to the `README.md`?

**Pull request merged**